File: cpp/estrutura_condicional/baskara.cpp
/* Problema "baskara"
 Fazer um programa para ler os três coeficientes de uma equação do segundo grau. Usando a fórmula
 de Baskara, calcular e mostrar os valores das raízes x1 e x2 da equação com quatro casas decimais,
 conforme exemplo. Se a equação não possuir raízes reais, mostrar uma mensagem.
*/

#include <cmath>
#include <iomanip>
#include <iostream>

using namespace std;

int main()
{
  double a, b, c, delta, x1, x2;

  cout << "Coeficiente a: ";
  cin >> a;

  cout << "Coeficiente b: ";
  cin >> b;

  cout << "Coeficiente c: ";
  cin >> c;

  delta = b * b - 4 * a * c;

  if (delta >= 0 && a != 0)
  {
    x1 = (-b + sqrt(delta)) / (2 * a);
    x2 = (-b - sqrt(delta)) / (2 * a);

    cout << fixed << setprecision(4);
    cout << "X1 = " << x1 << endl;
    cout << "X2 = " << x2 << endl;
  }
  else
  {
    cout << "Esta equação não possui raízes reais\n";
  }

  return 0;
}

File: csharp/Source/Estruturas/Condicional/Baskara.cs
namespace Source.Estruturas.Condicional;

/**
 * Problema "baskara"
 * Fazer um programa para ler os três coeficientes de uma equação do segundo grau.
 * Usando a fórmula de Baskara, calcular e mostrar os valores das raízes x1 e x2
 * da equação com quatro casas decimais. Se a equação não possuir raízes reais,
 * mostrar uma mensagem.
 */

internal static class Baskara
{
    public static void Execute()
    {

        Console.Write("Coeficiente a: ");
        double a = double.Parse(Console.ReadLine()!);

        Console.Write("Coeficiente b: ");
        double b = double.Parse(Console.ReadLine()!);

        Console.Write("Coeficiente c: ");
        double c = double.Parse(Console.ReadLine()!);

        double delta = b * b - 4 * a * c;

        if (delta >= 0 && a != 0)
        {
            double x1 = (-b + Math.Sqrt(delta)) / (2 * a);
            double x2 = (-b - Math.Sqrt(delta)) / (2 * a);

            Console.WriteLine("X1 = {0:F4}", x1);
            Console.WriteLine("X2 = {0:F4}", x2);
        }
        else
        {
            Console.WriteLine("Esta equação não possui raízes reais");
        }
    }
}